### PR TITLE
#14145: Update Documentation for backward ops

### PR DESCRIPTION
**Pull request by mouliraj-mcw**
### Ticket
Link to Github Issue #14144 , #14145 , #14293

### What's changed
Updated the documentation 

### Checklist
- [x] All Post commit CI

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp [hunk 1]
```diff
@@ -39,6 +39,8 @@ void bind_complex_unary_backward(py::module& module, const complex_unary_backwar
         Returns:
             List of ttnn.Tensor: the output tensor.
 
+        Note:
+            {3}
```
Comment (mcw-anasuya) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp:43]: You can remove memory config defaults to None. It is not mentioned in the arguments

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp [hunk 2]
```diff
@@ -88,6 +91,8 @@ void bind_complex_unary_backward_tensor(py::module& module, const complex_unary_
         Returns:
             List of ttnn.Tensor: the output tensor.
 
+        Note:
```
Comment (mcw-anasuya) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp:94]: You can remove defaults to None here too

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp [hunk 3]
```diff
@@ -125,7 +131,16 @@ void py_module(py::module& module) {
...
         module,
         ttnn::polar_bw,
-        R"doc(Performs backward operations for complex polar function on :attr:`input_tensor` with given :attr:`grad_tensor`.)doc");
+        R"doc(Performs backward operations for complex polar function on :attr:`input_tensor` with given :attr:`grad_tensor`.)doc",
+        R"doc(Supported dtypes, layouts, and ranks:
+
+           +----------------------------+---------------------------------+-------------------+
+           |     Dtypes                 |         Layouts                 |     Ranks         |
+           +----------------------------+---------------------------------+-------------------+
+           |    BFLOAT16, BFLOAT8_B     |       TILE                      |      2, 3, 4      |
```
Comment (mcw-anasuya) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp:140]: The issue says bfloat8_b is not supported?

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp [hunk 4]
```diff
@@ -145,7 +160,16 @@ void py_module(py::module& module) {
...
         module,
         ttnn::angle_bw,
-        R"doc(Performs backward operations for complex angle function on :attr:`input_tensor` with given :attr:`grad_tensor`.)doc");
+        R"doc(Performs backward operations for complex angle function on :attr:`input_tensor` with given :attr:`grad_tensor`.)doc",
+        R"doc(Supported dtypes, layouts, and ranks:
+
+           +----------------------------+---------------------------------+-------------------+
+           |     Dtypes                 |         Layouts                 |     Ranks         |
+           +----------------------------+---------------------------------+-------------------+
+           |    BFLOAT16, BFLOAT8_B     |       TILE                      |      2, 3, 4      |
```
Comment (mcw-anasuya) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp:169]: Here too the issue says bfloat8_b is not supported. Can you confirm by running the sweep tests?

File: ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp [hunk 5]
```diff
@@ -66,11 +69,11 @@ void bind_complex_unary_backward(py::module& module, const complex_unary_backwar
             py::arg("grad_tensor"),
             py::arg("input_tensor"),
             py::kw_only(),
-            py::arg("memory_config")});
+            py::arg("memory_config") = std::nullopt});
```
Comment (mcw-anasuya) [inline ttnn/cpp/ttnn/operations/eltwise/complex_unary_backward/complex_unary_backward_pybind.hpp:72]: Remove std::null opt

**Review by mcw-anasuya — APPROVED** [review]

**Review by eyonland — APPROVED** [review]

**Pull request merged**